test/libsolidity/solidityExecutionFramework.h

```diff
@@ -170,6 +173,22 @@ class ExecutionFramework
 	{
 		return encodeArgs(u256(0x20), u256(_arg.size()), _arg);
 	}
+	static u256 fixed(rational const& _value, int _fixedBits)
+	{
+		rational maxValue = (_value < 0) ? 		
+			rational(bigint(1) << 255, 1):
+			rational((bigint(1) << 256) - 1, 1);
+		unsigned convertingBits = 0;
+		rational value = _value;
+		while (value * 0x100 <= maxValue && value.denominator() != 1 && convertingBits < 256)
```

Comment:
This seems over-complicated. If you specify the number of fractional bits you want as part of the input, this is just
```
rational value = _value * boost::multiprecision::pow(bigint(2), _fractionalBits);
return u256(value.numerator() / value.denominator());
```
```